From 4713a2de0ce7cea7aba82b10ad8bdd9cfb63d3fa Mon Sep 17 00:00:00 2001
From: Dheerendra Purohit <dheerendra@pthinks.com>
Date: Fri, 4 Jul 2025 16:10:53 +0530
Subject: [PATCH 062/239] Doc: Add QML docs for Window::startSystemMove and
 startSystemResize
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

startSystemMove() and startSystemResize() are callable from QML but
were undocumented.
This change adds \qmlmethod documentation under the Window type.

Pick-to: 6.9 6.8
Fixes: QTBUG-126558
Change-Id: Ib34a25fa3f9e12cc6fab85732c852a81448dcc4b
Reviewed-by: Tor Arne Vestbø <tor.arne.vestbo@qt.io>
Reviewed-by: Safiyyah Moosa <safiyyah.moosa@qt.io>
(cherry picked from commit 775861c9cacb4e3cb10ffce1a5844e6bdfc53307)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/quick/items/qquickwindow.cpp | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/src/quick/items/qquickwindow.cpp b/src/quick/items/qquickwindow.cpp
index f1b131e7e9..eecfafc67b 100644
--- a/src/quick/items/qquickwindow.cpp
+++ b/src/quick/items/qquickwindow.cpp
@@ -1129,6 +1129,37 @@ void QQuickWindowPrivate::cleanup(QSGNode *n)
     QQuickGraphicsDevice, QQuickGraphicsConfiguration, QSGRendererInterface
 */
 
+/*!
+    \qmlmethod void Window::startSystemMove()
+    \since 6.8
+
+    \brief Starts a system-specific move operation.
+
+    Starts an interactive move operation on the window using platform support.
+    The window follows the mouse cursor until the mouse button is released.
+
+    Use this method instead of \c setPosition, because it allows the window manager
+    to handle snapping, tiling, and related animations. On Wayland, \c setPosition
+    is not supported, so this is the only way the application can influence the
+    window’s position.
+*/
+
+/*!
+    \qmlmethod void Window::startSystemResize(Qt::Edges edges)
+    \since 6.8
+
+    \brief Starts a system-specific resize operation.
+
+    Starts an interactive resize operation on the window using platform support.
+    The specified edge follows the mouse cursor while dragging.
+
+    Use this method instead of \c setGeometry, because it allows the window manager
+    to handle snapping and resize animations when resizing to screen edges.
+
+    \a edges must be a single edge or a combination of two adjacent edges (a corner).
+    Other values are not allowed.
+*/
+
 /*!
     Constructs a window for displaying a QML scene with parent window \a parent.
 */
-- 
2.51.2

